File: Client/app/src/main/java/com/shortdrama/app/data/model/Episode.kt
package com.shortdrama.app.data.model

/**
 *
 * 后端路由: GET /api/episodes（列表） / GET /api/episodes/:id（详情）
 *
 * 列表接口返回的 highlights = []（后端 omitempty）
 * 详情接口返回的 highlights = 该集所有高光事件
 *
 * duration 单位秒；video_url 可直接传给 ExoPlayer/Media3 播放。
 */
data class Episode(
    val id: Long,
    val drama_id: Long,
    val episode_no: Int,
    val video_url: String,
    val duration: Long,
    val highlights: List<Highlight> = emptyList()
)

File: Backend/internal/model/episode.go
package model

type Episode struct {
	ID         uint64           `json:"id" gorm:"primaryKey;autoIncrement"`
	DramaID    uint64           `json:"drama_id" gorm:"not null;uniqueIndex:idx_drama_episode_no"`
	EpisodeNo  int              `json:"episode_no" gorm:"not null;uniqueIndex:idx_drama_episode_no"`
	VideoURL   string           `json:"video_url" gorm:"size:500;not null"`
	Duration   int              `json:"duration" gorm:"not null"`
	Highlights []HighlightEvent `json:"highlights,omitempty" gorm:"constraint:OnUpdate:CASCADE,OnDelete:CASCADE;"`
}

func (Episode) TableName() string {
	return "episode"
}
